templates/CRM/Event/Form/Registration/DisplayProfile.tpl

```diff
@@ -12,30 +12,59 @@
         <div class="crm-group participant_info-group">
       <div class="header-dark">{if $addParticipantProfile}{ts}Participant 1{/ts}{else}{ts}Participant Information{/ts}{/if}</div>
             {if $primaryParticipantProfile.CustomPre}
-               <fieldset class="label-left no-border"><div class="bold crm-profile-view-title">{$primaryParticipantProfile.CustomPreGroupTitle}</div>
+              {assign var="hasValue" value=false}
+              {foreach from=$primaryParticipantProfile.CustomPre item=value key=field}
+                {if $value !== ''}
+                  {assign var="hasValue" value=true}
+                  {break} <!-- exit if a value is found -->
+                {/if}
+              {/foreach}
+
+              {if $hasValue}
+                <fieldset class="label-left no-border">
+                   <div class="bold crm-profile-view-title">{$primaryParticipantProfile.CustomPreGroupTitle}</div>
                    {foreach from=$primaryParticipantProfile.CustomPre item=value key=field}
-                      <div class="crm-public-form-item crm-section {$field}-section">
-                          <div class="label">{$field}</div>
-                          <div class="content">{$value}</div>
-                          <div class="clear"></div>
-                      </div>
+                      {if $value !== ''}
+                        <div class="crm-public-form-item crm-section {$field}-section">
+                            <div class="label">{$field}</div>
+                            <div class="content">{$value}</div>
+                            <div class="clear"></div>
+                        </div>
+                      {/if}
                    {/foreach}
-               </fieldset>
+                </fieldset>
+              {/if}
             {/if}
-         {if array_key_exists('CustomPost', $primaryParticipantProfile) && $primaryParticipantProfile.CustomPost}
-               {foreach from=$primaryParticipantProfile.CustomPost item=value key=field}
-                  <fieldset class="label-left no-border"><div class="bold crm-profile-view-title">{$primaryParticipantProfile.CustomPostGroupTitle.$field.groupTitle}</div>
+
+            {if array_key_exists('CustomPost', $primaryParticipantProfile) && $primaryParticipantProfile.CustomPost}
+              {assign var="hasValue" value=false}
+              {foreach from=$primaryParticipantProfile.CustomPost item=fieldValues key=field}
+                {foreach from=$fieldValues item=value}
+                  {if $value !== ''}
+                    {assign var="hasValue" value=true}
+                    {break 2} <!-- exit both loops if a non-empty value is found -->
```

Comment:
You can have multiple profiles configured on the event, so you'd need to check each group separately, otherwise for example if I enter a value for a field in the first group, but not in the second, it will still display the group title for the second group.